### PR TITLE
continution to #1708

### LINKED ISSUE #1503: [Concept Entry] Go: Interfaces
### Reason/inspiration (optional)

We would like a new term entry for [Go](https://www.codecademy.com/resources/docs/go) about Interfaces.

The entry should go in a new file in docs/content/go/concepts/interfaces/interfaces.md

The entry should

Describe what interfaces are and their relationship to types.
Show a type implementing an interface.
Have a syntax section illustrating an interface structure with a \`\`\`pseudo code block.
Have an example showing an interface in use with a \`\`\`go code block.
Have a runnable codebyte example in a \`\`\`codebyte/golang block.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Pavan-04-08**
### Description

coninution to #1708  closes #1503  refer https://github.com/Codecademy/docs/pull/1708#issuecomment-1504719397 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Pavan-04-08** [comment]
review 1 and 2 completed a review check shud be left by @yangc95  as i made new pr i would like to address concerns here first before commiting

**Comment by Pavan-04-08** [comment]
correct me if i am wrong here things this PR should address which are not yet addressed in old PR

> I think another issue here is that repeating the same examples (but with different names e.g., John and max) don't provide additional value or any new information here.

**Comment by Pavan-04-08** [comment]
and so before proceeding is this approved? https://github.com/Codecademy/docs/pull/1708#discussion_r1162217190

**Comment by yangc95** [comment]
@Pavan-04-08 If you're unsure how to proceed, let me know. Otherwise, please address the previous comment that was quoted. Thank you

**Comment by Pavan-04-08** [comment]
> @Pavan-04-08 If you're unsure how to proceed, let me know. Otherwise, please address the previous comment that was quoted. Thank you

um yeah i understand tht but 




> and so before proceeding is this approved? [#1708 (comment)](https://github.com/Codecademy/docs/pull/1708#discussion_r1162217190)
> 
> if this is approved i should do some refactoring like changing it into diff files n all so

this approved or not? coz if this is approved i have to change things like dividing interfaces into diff chunks like arraylist methods link i provided for reference

if this is not ok i will just edit example as u said n that should be good to go

**Comment by yangc95** [comment]
@Pavan-04-08 Please edit the example, after which I can approve of this PR, and it can be merged.

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:195]:
```suggestion
```

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:193]:
```suggestion
A simple example on how to use interfaces, `square` and `circle` both implement the `geometry` interface as both implement the required methods.
```

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:201]:
```suggestion
In the code below, the `geometry` interface has two methods `area()` and `perimeter()`, both of which return a float64. `square` and `circle` implement these methods, allowing them to be used interchangeably as geometric objects.

The code then defines a `printGeometry()` function, which takes a geometry object as its parameter and prints its details, including its area and perimeter.

In the `main()` function, a square and a circle are created, and the `printGeometry()` function is called for each of them, demonstrating how the two different types can be used interchangeably as geometric objects.
```

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:202]:
```suggestion
```suggestion
```
```

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:249]:
```suggestion
```

File: content/go/concepts/interfaces/interfaces.md
Comment (yangc95) [inline content/go/concepts/interfaces/interfaces.md:167]:
```suggestion
    fmt.Println("\n", l.quote)
```

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@Pavan-04-08 live link: https://www.codecademy.com/resources/docs/go/interfaces

Screenshot 2023-04-25 at 4 16 33 PM

cc: @KTom101 @SSwiniarski

**Pull request merged**
